### ISSUE TITLE
diy doen't generate existence conditions for memory locations with memtag

### ISSUE THREAD

**Issue by ambroise-arm** [herd/herdtools7#68]
Result of `diyone7 -arch AArch64 Rfe PodRW Coe PodWW`:
```
Orig=Rfe PodRW Coe PodWW
{
0:X1=x; 0:X3=y;
1:X1=y; 1:X3=x;
}
 P0          | P1          ;
 LDR W0,[X1] | MOV W0,#2   ;
 MOV W2,#1   | STR W0,[X1] ;
 STR W2,[X3] | MOV W2,#1   ;
             | STR W2,[X3] ;
exists (y=2 /\ 0:X0=1)
```

Result of `diyone7 -arch AArch64 Rfe PodRW CoeTT PodWW -variant memtag`:
```
Orig=Rfe PodRWPT CoeTT PodWWTP
{
0:X1=x:green; 0:X2=y:red; 0:X3=y:green;
1:X0=y:blue; 1:X1=y:red; 1:X3=x:green;
}
 P0          | P1          ;
 LDR W0,[X1] | STG X0,[X1] ;
 STG X2,[X3] | MOV W2,#1   ;
             | STR W2,[X3] ;
exists (0:X0=1)
```

The existence condition of the second litmus should mention that "y" is expected to be blue.

**Comment by maranget** [comment]
Hi, this would mean checking the color once the test is over, this is possible, although not a priority at the moment. Feel free to submit a pull request on this topic.

**Comment by jalglave** [comment]
hi Luc,

this also occurs in normal mode, not just for memtag. The behaviour in
normal mode is not right I think.
Thanks,
Jade

**Comment by maranget** [comment]
Issue solved by PR #269.
